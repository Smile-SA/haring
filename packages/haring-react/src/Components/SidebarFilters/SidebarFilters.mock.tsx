import type { IMenuItem } from '../SidebarMenu/SidebarMenu';

import { Checkbox, Group } from '@mantine/core';

import { SearchableList } from '../SearchableList/SearchableList';

export const activeFilters = [
  {
    categoryId: [1],
    id: 1,
    label: 'Dupont',
    value: 'DUPONT',
  },
  {
    categoryId: [1],
    id: 2,
    label: 'Martin',
    value: 'MARTIN',
  },
  {
    categoryId: [3, 10],
    id: 3,
    label: 'Freelance',
    value: 'Freelance',
  },
  {
    categoryId: [3, 11],
    id: 4,
    label: 'CDI',
    value: 'CDI',
  },
];

export function getMenu(checked: boolean): IMenuItem<number>[] {
  return [
    {
      content: (
        <>
          <Group>
            <Checkbox defaultChecked={checked} value="Dupont" /> Dupont
          </Group>
          <br />
          <Group>
            <Checkbox defaultChecked={checked} value="Martin" /> Martin
          </Group>
          <br />
          <Group>
            <Checkbox value="André" /> Andrée
          </Group>
        </>
      ),
      id: 1,
      label: 'Client name',
    },
    {
      children: [
        {
          content: (
            <>
              <Group>
                <Checkbox defaultChecked={checked} value="CDI" /> CDI
              </Group>
              <br />
              <Group>
                <Checkbox value="CDD" /> CDD
              </Group>
            </>
          ),
          id: 10,
          label: 'Classic contract',
        },
        {
          content: (
            <Group>
              <Checkbox defaultChecked={checked} value="FREELANCE" /> Freelance
            </Group>
          ),
          id: 11,
          label: 'Special contract',
        },
      ],
      id: 3,
      label: 'Type of Contract',
    },
    {
      id: 8,
      label: 'Period',
    },
    { id: 9, label: 'Contract duration' },
  ];
}

const clientsMock = [
  { active: true, id: 1, label: 'Dupont', value: 'dupont' },
  { active: true, id: 2, label: 'Martin', value: 'martin' },
  { active: false, id: 5, label: 'Andrée', value: 'andree' },
];

const classicsContractsMock = [
  { active: true, id: 4, label: 'CDI', value: 'cdi' },
  { active: false, id: 7, label: 'CDD', value: 'cdd' },
];

const specialContractMock = [
  { active: true, id: 3, label: 'Freelance', value: 'freelance' },
];

export function getMenuWithSelectableList(): IMenuItem<number>[] {
  return [
    {
      content: (
        <SearchableList
          checkboxes={clientsMock}
          minSearchableItems={0}
          submitButton={false}
        />
      ),
      id: 1,
      label: 'Client name',
    },
    {
      children: [
        {
          content: (
            <SearchableList
              checkboxes={classicsContractsMock}
              minSearchableItems={0}
              submitButton={false}
            />
          ),
          id: 10,
          label: 'Classic contract',
        },
        {
          content: (
            <SearchableList
              checkboxes={specialContractMock}
              minSearchableItems={0}
              submitButton={false}
            />
          ),
          id: 11,
          label: 'Special contract',
        },
      ],
      id: 3,
      label: 'Type of Contract',
    },
    {
      id: 8,
      label: 'Period',
    },
    { id: 9, label: 'Contract duration' },
  ];
}
