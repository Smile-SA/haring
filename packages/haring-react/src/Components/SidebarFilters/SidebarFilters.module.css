.activeFilters {
  flex-wrap: wrap;
  margin-top: 16px;
}

.activeFiltersButtonRoot {
  width: 100%;
}

.activeFiltersCollapseInner {
  display: flex;
  justify-content: space-between;
}

.activeFiltersCollapseLabel {
  color: var(--mantine-color-gray-0);
  text-align: left;
}

.activeFiltersCollapseRoot {
  background: transparent;
  padding: 0;
  &:hover {
    background-color: transparent;
  }
}

.badgeInner {
  &::first-letter {
    text-transform: uppercase;
  }

  font-weight: 600;
  position: relative;
  right: 4px;
  text-transform: initial;
}

.badgeRight {
  position: relative;
  right: 4px;
  top: 1px;
}

.badgeRoot {
  border: 2px solid var(--mantine-color-white);
  color: var(--mantine-color-white);
  cursor: pointer;
  font-size: 12px;
  font-weight: normal;
  margin-bottom: 8px;
  margin-right: 8px;
  min-width: 60px;
}

.bottom {
  padding: 24px 24px 48px 24px;
}

.controlledMenu {
  border-bottom: var(--mantine-color-gray-3) 1px solid;
  color: var(--mantine-color-dark-3);
  font-weight: 600;
  padding-bottom: 10px;
}

.controlledMenuButton {
  cursor: pointer;
}

.controlledMenuLine {
  background: var(--mantine-color-gray-2);
  height: 25px;
  width: 1px;
}

.middle {
  margin: 0 24px;
}

.removeAllFiltersButtonRoot {
  margin-top: 8px;
  width: 100%;
}

.root {
  background-color: var(--mantine-color-white);
  border-radius: 16px;
}

.sidebarMenuButtonInner {
  justify-content: Left;
}

.sidebarMenuButtonLabel {
  color: var(--mantine-color-dark-filled);
  text-align: left;
}

.sidebarMenuButtonRoot {
  &[data-selected='false'] {
    background: inherit;
  }

  background: var(--mantine-primary-color-0);
}

.sidebarMenuContentContainer {
  padding: 10px;
}

.title {
  font-weight: 600;
}

.top {
  background: var(--mantine-primary-color-filled);
  border: 8px solid var(--mantine-color-white);
  border-radius: 16px 16px 0 0;
  color: var(--mantine-color-white);
  margin-bottom: 24px;
  padding: 24px 44px;

  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    padding: 24px 16px;
  }
}
