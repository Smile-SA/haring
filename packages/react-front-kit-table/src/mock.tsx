/* eslint-disable react-refresh/only-export-components */
export * from './Components/Table/Table.mock';
export * from './Components/TableGridView/TableGridView.mock';
