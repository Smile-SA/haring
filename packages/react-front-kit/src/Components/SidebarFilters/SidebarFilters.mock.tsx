import type { IMenuItem } from '../SidebarMenu/SidebarMenu';

import { Checkbox, Group } from '@mantine/core';

export const activeFilters = [
  {
    categoryId: [1],
    id: 1,
    label: 'Dupont',
    value: 'DUPONT',
  },
  {
    categoryId: [1],
    id: 2,
    label: 'Martin',
    value: 'MARTIN',
  },
  {
    categoryId: [3, 10],
    id: 3,
    label: 'Freelance',
    value: 'Freelance',
  },
  {
    categoryId: [3, 11],
    id: 4,
    label: 'CDI',
    value: 'CDI',
  },
];

export function getMenu(checked: boolean): IMenuItem<number>[] {
  return [
    {
      content: (
        <>
          <Group>
            <Checkbox defaultChecked={checked} value="Dupont" /> Dupont
          </Group>
          <br />
          <Group>
            <Checkbox defaultChecked={checked} value="Martin" /> Martin
          </Group>
          <br />
          <Group>
            <Checkbox value="André" /> Andrée
          </Group>
        </>
      ),
      id: 1,
      label: 'Nom du client',
    },
    {
      children: [
        {
          content: (
            <>
              <Group>
                <Checkbox defaultChecked={checked} value="CDI" /> CDI
              </Group>
              <br />
              <Group>
                <Checkbox value="CDD" /> CDD
              </Group>
            </>
          ),
          id: 10,
          label: 'Contrat classique',
        },
        {
          content: (
            <Group>
              <Checkbox defaultChecked={checked} value="FREELANCE" /> Freelance
            </Group>
          ),
          id: 11,
          label: 'Contrat special',
        },
      ],
      id: 3,
      label: 'Type de contrat',
    },
    {
      id: 8,
      label: 'Période',
    },
    { id: 9, label: 'Durée du contrat' },
  ];
}
