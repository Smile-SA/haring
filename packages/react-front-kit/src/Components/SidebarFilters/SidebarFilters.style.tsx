import { createStyles } from '@mantine/styles';

export const useStyles = createStyles((theme) => ({
  activeFilters: {
    flexWrap: 'wrap',
    marginTop: '16px',
  },
  activeFiltersButtonRoot: {
    minWidth: '100%',
    paddingRight: '0',
  },
  activeFiltersCollapseInner: {
    display: 'flex',
    justifyContent: 'space-between',
  },
  activeFiltersCollapseLabel: {
    color: theme.colors.gray[0],
    textAlign: 'left',
  },
  activeFiltersCollapseRoot: {
    background: 'inherit',
    padding: '0px',
  },
  badgeInner: {
    '&::first-letter': {
      textTransform: 'uppercase',
    },
    fontWeight: 600,
    position: 'relative',
    right: '4px',
    textTransform: 'initial',
  },
  badgeRight: {
    position: 'relative',
    right: '4px',
    top: '1px',
  },
  badgeRoot: {
    border: `2px solid ${theme.white}`,
    color: theme.white,
    cursor: 'pointer',
    fontSize: '12px',
    fontWeight: 'normal',
    marginBottom: '8px',
    marginRight: '8px',
    minWidth: '60px',
  },
  bottom: {
    padding: '24px 24px 48px 24px',
  },
  controlledMenu: {
    borderBottom: `${theme.colors.gray[3]} 1px solid`,
    color: theme.colors.dark[3],
    fontWeight: 600,
    paddingBottom: '10px',
  },
  controlledMenuButton: {
    cursor: 'pointer',
  },
  controlledMenuLine: {
    background: theme.colors.gray[2],
    height: '25px',
    width: '1px',
  },
  middle: {
    margin: '0px 24px',
  },
  removeAllFiltersButtonRoot: {
    marginTop: '8px',
    width: '100%',
  },
  root: {
    backgroundColor: theme.white,
    borderRadius: '16px',
  },
  sidebarMenuButtonInner: {
    justifyContent: 'Left',
  },
  sidebarMenuButtonLabel: {
    color: theme.colors.dark,
    textAlign: 'left',
  },
  sidebarMenuButtonRoot: {
    '&[data-selected="false"]': {
      background: 'inherit',
    },
    background: theme.colors[theme.primaryColor][0],
  },
  sidebarMenuContentContainer: {
    padding: '10px',
  },
  title: {
    fontWeight: 600,
  },
  top: {
    [`@media (max-width: ${theme.breakpoints.sm})`]: {
      padding: '24px 16px',
    },
    background: theme.fn.primaryColor(),
    border: `8px solid ${theme.white}`,
    borderRadius: '16px 16px 0px 0px',
    color: theme.white,
    marginBottom: '24px',
    padding: '24px 44px',
  },
}));
