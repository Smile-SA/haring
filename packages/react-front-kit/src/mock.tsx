/* eslint-disable react-refresh/only-export-components */

export * from './Components/ResponsiveTabs/ResponsiveTabs.mock';
export * from './Components/SidebarMenu/SidebarMenu.mock';
export * from './Components/SwitchableView/SwitchableView.mock';
export * from './Components/Thumbnail/Thumbnail.mock';
export * from './Components/ThumbnailGrid/ThumbnailGrid.mock';
export * from './Layouts/FoldableColumnLayout/FoldableColumnLayout.mock';
