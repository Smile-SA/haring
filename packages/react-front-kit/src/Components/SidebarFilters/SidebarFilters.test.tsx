import { renderWithProviders } from '@smile/react-front-kit-shared/test-utils';

import { SidebarFilters } from './SidebarFilters';
import { activeFilters, getMenu } from './SidebarFilters.mock';

describe('SidebarFilters', () => {
  beforeEach(() => {
    // Prevent mantine random ID
    Math.random = () => 0.42;
  });
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <SidebarFilters
        activeFilters={activeFilters}
        defaultOpenedMenuIds={[1, 3, 10, 11]}
        deleteButtonLabel="Remove all"
        filterButtonLabel="Filtrer"
        menus={getMenu(true)}
        title="Active filters"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
