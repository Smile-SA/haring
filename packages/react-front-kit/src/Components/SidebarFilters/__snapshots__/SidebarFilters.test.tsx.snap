// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarFilters matches snapshot 1`] = `
<div>
  <div
    class="mantine-1qn0wrz"
  >
    <div
      class="mantine-gwowuu"
    >
      <div
        aria-expanded="true"
        class="mantine-UnstyledButton-root mantine-Button-root mantine-1fajwen"
        data-button="true"
        data-testid="root"
        type="button"
      >
        <div
          class="mantine-Button-inner mantine-uf8iid"
        >
          <span
            class="mantine-Button-label mantine-28zsh7"
          >
            <button
              class="mantine-pax7vz "
              data-testid="select"
              type="button"
            >
              <span
                class="mantine-l9up18"
              >
                Active filters
                 (
                4
                )
              </span>
            </button>
          </span>
          <span
            class="mantine-Button-icon mantine-Button-rightIcon mantine-467wo8"
          >
            <button
              class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-gj6ulq"
              data-testid="toggle"
              type="button"
            >
              <svg
                fill="white"
                height="1em"
                viewBox="0 0 256 256"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M213.66,165.66a8,8,0,0,1-11.32,0L128,91.31,53.66,165.66a8,8,0,0,1-11.32-11.32l80-80a8,8,0,0,1,11.32,0l80,80A8,8,0,0,1,213.66,165.66Z"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
      <div
        aria-hidden="false"
        class="mantine-1avyp1d"
        data-testid="content"
        style="box-sizing: border-box;"
      >
        <div
          style="opacity: 1; transition: opacity 200ms ease;"
        >
          <div
            class="mantine-fi6uov"
          >
            <div
              class="mantine-Badge-root mantine-ogj4ps"
            >
              <span
                class="mantine-Badge-inner mantine-ucd2eg"
              >
                Dupont
              </span>
              <span
                class="mantine-Badge-rightSection mantine-4aag21"
              >
                <svg
                  fill="currentColor"
                  height="10"
                  viewBox="0 0 256 256"
                  width="10"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
                  />
                </svg>
              </span>
            </div>
            <div
              class="mantine-Badge-root mantine-ogj4ps"
            >
              <span
                class="mantine-Badge-inner mantine-ucd2eg"
              >
                Martin
              </span>
              <span
                class="mantine-Badge-rightSection mantine-4aag21"
              >
                <svg
                  fill="currentColor"
                  height="10"
                  viewBox="0 0 256 256"
                  width="10"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
                  />
                </svg>
              </span>
            </div>
            <div
              class="mantine-Badge-root mantine-ogj4ps"
            >
              <span
                class="mantine-Badge-inner mantine-ucd2eg"
              >
                Freelance
              </span>
              <span
                class="mantine-Badge-rightSection mantine-4aag21"
              >
                <svg
                  fill="currentColor"
                  height="10"
                  viewBox="0 0 256 256"
                  width="10"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
                  />
                </svg>
              </span>
            </div>
            <div
              class="mantine-Badge-root mantine-ogj4ps"
            >
              <span
                class="mantine-Badge-inner mantine-ucd2eg"
              >
                CDI
              </span>
              <span
                class="mantine-Badge-rightSection mantine-4aag21"
              >
                <svg
                  fill="currentColor"
                  height="10"
                  viewBox="0 0 256 256"
                  width="10"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mantine-17q7vz3"
    >
      <div
        class="mantine-Group-root mantine-16g550v"
      >
        <span
          aria-hidden="true"
          class="mantine-134spwq"
        >
          - Close all
        </span>
        <span
          class="mantine-5hbfsj"
        />
        <span
          aria-hidden="true"
          class="mantine-134spwq"
        >
          + Open all
        </span>
      </div>
      <div
        class="mantine-Paper-root mantine-1lnkm2l"
      >
        <div
          aria-expanded="true"
          class="mantine-UnstyledButton-root mantine-Button-root mantine-6dql8f"
          data-button="true"
          data-selected="false"
          data-testid="root"
          type="button"
        >
          <div
            class="mantine-Button-inner mantine-1b27hci"
          >
            <span
              class="mantine-Button-label mantine-m141sh"
            >
              <button
                class="mantine-pax7vz "
                data-testid="select"
                type="button"
              >
                Nom du client (2)
              </button>
            </span>
            <span
              class="mantine-Button-icon mantine-Button-rightIcon mantine-467wo8"
            >
              <button
                class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-gj6ulq"
                data-testid="toggle"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
                  />
                </svg>
              </button>
            </span>
          </div>
        </div>
        <div
          aria-hidden="false"
          class="mantine-16d64p7"
          data-testid="content"
          style="box-sizing: border-box; border-top: 1px solid #dee2e6;"
        >
          <div
            style="opacity: 1; transition: opacity 200ms ease;"
          >
            <div
              class="mantine-kz5t98"
            >
              <div
                class="mantine-Group-root mantine-k3ov3c"
              >
                <div
                  class="mantine-Checkbox-root mantine-yxmaw9"
                >
                  <div
                    class="mantine-hqx0d1 mantine-Checkbox-body"
                  >
                    <div
                      class="mantine-1vf457v mantine-Checkbox-inner"
                    >
                      <input
                        checked=""
                        class="mantine-1mibt2f mantine-Checkbox-input"
                        id="mantine-f4bipx4bi"
                        type="checkbox"
                        value="Dupont"
                      />
                      <svg
                        class="___ref-icon mantine-q9qwbl mantine-Checkbox-icon"
                        fill="none"
                        viewBox="0 0 10 7"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                    <div
                      class="mantine-g88pth mantine-Checkbox-labelWrapper"
                    />
                  </div>
                </div>
                 Dupont
              </div>
              <br />
              <div
                class="mantine-Group-root mantine-k3ov3c"
              >
                <div
                  class="mantine-Checkbox-root mantine-yxmaw9"
                >
                  <div
                    class="mantine-hqx0d1 mantine-Checkbox-body"
                  >
                    <div
                      class="mantine-1vf457v mantine-Checkbox-inner"
                    >
                      <input
                        checked=""
                        class="mantine-1mibt2f mantine-Checkbox-input"
                        id="mantine-f4bipx4bi"
                        type="checkbox"
                        value="Martin"
                      />
                      <svg
                        class="___ref-icon mantine-q9qwbl mantine-Checkbox-icon"
                        fill="none"
                        viewBox="0 0 10 7"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                    <div
                      class="mantine-g88pth mantine-Checkbox-labelWrapper"
                    />
                  </div>
                </div>
                 Martin
              </div>
              <br />
              <div
                class="mantine-Group-root mantine-k3ov3c"
              >
                <div
                  class="mantine-Checkbox-root mantine-yxmaw9"
                >
                  <div
                    class="mantine-hqx0d1 mantine-Checkbox-body"
                  >
                    <div
                      class="mantine-1vf457v mantine-Checkbox-inner"
                    >
                      <input
                        class="mantine-1mibt2f mantine-Checkbox-input"
                        id="mantine-f4bipx4bi"
                        type="checkbox"
                        value="André"
                      />
                      <svg
                        class="___ref-icon mantine-q9qwbl mantine-Checkbox-icon"
                        fill="none"
                        viewBox="0 0 10 7"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                    <div
                      class="mantine-g88pth mantine-Checkbox-labelWrapper"
                    />
                  </div>
                </div>
                 Andrée
              </div>
            </div>
          </div>
        </div>
        <div
          aria-expanded="true"
          class="mantine-UnstyledButton-root mantine-Button-root mantine-6dql8f"
          data-button="true"
          data-selected="false"
          data-testid="root"
          type="button"
        >
          <div
            class="mantine-Button-inner mantine-1b27hci"
          >
            <span
              class="mantine-Button-label mantine-m141sh"
            >
              <button
                class="mantine-pax7vz "
                data-testid="select"
                type="button"
              >
                Type de contrat (2)
              </button>
            </span>
            <span
              class="mantine-Button-icon mantine-Button-rightIcon mantine-467wo8"
            >
              <button
                class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-gj6ulq"
                data-testid="toggle"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
                  />
                </svg>
              </button>
            </span>
          </div>
        </div>
        <div
          aria-hidden="false"
          class="mantine-16d64p7"
          data-testid="content"
          style="box-sizing: border-box; border-top: 1px solid #dee2e6;"
        >
          <div
            style="opacity: 1; transition: opacity 200ms ease;"
          >
            <div
              aria-expanded="true"
              class="mantine-UnstyledButton-root mantine-Button-root mantine-6dql8f"
              data-button="true"
              data-selected="false"
              data-testid="root"
              type="button"
            >
              <div
                class="mantine-Button-inner mantine-1b27hci"
              >
                <span
                  class="mantine-Button-label mantine-m141sh"
                >
                  <button
                    class="mantine-pax7vz "
                    data-testid="select"
                    type="button"
                  >
                    Contrat classique (1)
                  </button>
                </span>
                <span
                  class="mantine-Button-icon mantine-Button-rightIcon mantine-467wo8"
                >
                  <button
                    class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-gj6ulq"
                    data-testid="toggle"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 256 256"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
                      />
                    </svg>
                  </button>
                </span>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="mantine-16d64p7"
              data-testid="content"
              style="box-sizing: border-box; border-top: 1px solid #dee2e6;"
            >
              <div
                style="opacity: 1; transition: opacity 200ms ease;"
              >
                <div
                  class="mantine-kz5t98"
                >
                  <div
                    class="mantine-Group-root mantine-k3ov3c"
                  >
                    <div
                      class="mantine-Checkbox-root mantine-yxmaw9"
                    >
                      <div
                        class="mantine-hqx0d1 mantine-Checkbox-body"
                      >
                        <div
                          class="mantine-1vf457v mantine-Checkbox-inner"
                        >
                          <input
                            checked=""
                            class="mantine-1mibt2f mantine-Checkbox-input"
                            id="mantine-f4bipx4bi"
                            type="checkbox"
                            value="CDI"
                          />
                          <svg
                            class="___ref-icon mantine-q9qwbl mantine-Checkbox-icon"
                            fill="none"
                            viewBox="0 0 10 7"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              clip-rule="evenodd"
                              d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
                              fill="currentColor"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </div>
                        <div
                          class="mantine-g88pth mantine-Checkbox-labelWrapper"
                        />
                      </div>
                    </div>
                     CDI
                  </div>
                  <br />
                  <div
                    class="mantine-Group-root mantine-k3ov3c"
                  >
                    <div
                      class="mantine-Checkbox-root mantine-yxmaw9"
                    >
                      <div
                        class="mantine-hqx0d1 mantine-Checkbox-body"
                      >
                        <div
                          class="mantine-1vf457v mantine-Checkbox-inner"
                        >
                          <input
                            class="mantine-1mibt2f mantine-Checkbox-input"
                            id="mantine-f4bipx4bi"
                            type="checkbox"
                            value="CDD"
                          />
                          <svg
                            class="___ref-icon mantine-q9qwbl mantine-Checkbox-icon"
                            fill="none"
                            viewBox="0 0 10 7"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              clip-rule="evenodd"
                              d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
                              fill="currentColor"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </div>
                        <div
                          class="mantine-g88pth mantine-Checkbox-labelWrapper"
                        />
                      </div>
                    </div>
                     CDD
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-expanded="true"
              class="mantine-UnstyledButton-root mantine-Button-root mantine-6dql8f"
              data-button="true"
              data-selected="false"
              data-testid="root"
              type="button"
            >
              <div
                class="mantine-Button-inner mantine-1b27hci"
              >
                <span
                  class="mantine-Button-label mantine-m141sh"
                >
                  <button
                    class="mantine-pax7vz "
                    data-testid="select"
                    type="button"
                  >
                    Contrat special (1)
                  </button>
                </span>
                <span
                  class="mantine-Button-icon mantine-Button-rightIcon mantine-467wo8"
                >
                  <button
                    class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-gj6ulq"
                    data-testid="toggle"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 256 256"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
                      />
                    </svg>
                  </button>
                </span>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="mantine-16d64p7"
              data-testid="content"
              style="box-sizing: border-box; border-top: 1px solid #dee2e6;"
            >
              <div
                style="opacity: 1; transition: opacity 200ms ease;"
              >
                <div
                  class="mantine-kz5t98"
                >
                  <div
                    class="mantine-Group-root mantine-k3ov3c"
                  >
                    <div
                      class="mantine-Checkbox-root mantine-yxmaw9"
                    >
                      <div
                        class="mantine-hqx0d1 mantine-Checkbox-body"
                      >
                        <div
                          class="mantine-1vf457v mantine-Checkbox-inner"
                        >
                          <input
                            checked=""
                            class="mantine-1mibt2f mantine-Checkbox-input"
                            id="mantine-f4bipx4bi"
                            type="checkbox"
                            value="FREELANCE"
                          />
                          <svg
                            class="___ref-icon mantine-q9qwbl mantine-Checkbox-icon"
                            fill="none"
                            viewBox="0 0 10 7"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              clip-rule="evenodd"
                              d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
                              fill="currentColor"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </div>
                        <div
                          class="mantine-g88pth mantine-Checkbox-labelWrapper"
                        />
                      </div>
                    </div>
                     Freelance
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-expanded="false"
          class="mantine-UnstyledButton-root mantine-Button-root mantine-893cod"
          data-button="true"
          data-selected="false"
          data-testid="root"
          type="button"
        >
          <div
            class="mantine-Button-inner mantine-1b27hci"
          >
            <span
              class="mantine-Button-label mantine-m141sh"
            >
              <button
                class="mantine-pax7vz "
                data-testid="select"
                type="button"
              >
                Période
              </button>
            </span>
          </div>
        </div>
        <div
          aria-expanded="false"
          class="mantine-UnstyledButton-root mantine-Button-root mantine-893cod"
          data-button="true"
          data-selected="false"
          data-testid="root"
          type="button"
        >
          <div
            class="mantine-Button-inner mantine-1b27hci"
          >
            <span
              class="mantine-Button-label mantine-m141sh"
            >
              <button
                class="mantine-pax7vz "
                data-testid="select"
                type="button"
              >
                Durée du contrat
              </button>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mantine-njwlxd"
    >
      <button
        class="mantine-UnstyledButton-root mantine-Button-root mantine-13zc0zg"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-1ryt1ht mantine-Button-label"
          >
            Filtrer
             (
            4
            )
          </span>
        </div>
      </button>
      <button
        class="mantine-UnstyledButton-root mantine-Button-root mantine-1nlhzt"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-Button-icon mantine-Button-leftIcon mantine-1jbrfp"
          >
            <svg
              fill="currentColor"
              height="12"
              viewBox="0 0 256 256"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M216,48H40a8,8,0,0,0,0,16h8V208a16,16,0,0,0,16,16H192a16,16,0,0,0,16-16V64h8a8,8,0,0,0,0-16ZM192,208H64V64H192ZM80,24a8,8,0,0,1,8-8h80a8,8,0,0,1,0,16H88A8,8,0,0,1,80,24Z"
              />
            </svg>
          </span>
          <span
            class="mantine-1ryt1ht mantine-Button-label"
          >
            Remove all
             (
            4
            )
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
